src/client.ts

```diff
@@ -145,40 +229,59 @@ class StandaloneConnection implements Connection {
     return this._isClosed;
   }
   async execute(query: string): Promise<void> {
+    this._check_borrow();
     let connection = this._connection;
     if (!connection || connection.isClosed()) {
       connection = await this._reconnect();
     }
     return await connection.execute(query);
   }
   async query(query: string, args?: QueryArgs): Promise<Set> {
+    this._check_borrow();
     let connection = this._connection;
     if (!connection || connection.isClosed()) {
       connection = await this._reconnect();
     }
     return await connection.query(query, args);
   }
   async queryJSON(query: string, args?: QueryArgs): Promise<string> {
+    this._check_borrow();
```

Comment:
JS JITs don't inline functions that raise errors. So I'd guard with `if (this[BORROW]) {`.

Reply:
Done